test/compilable/test22626.d

```diff
@@ -0,0 +1,16 @@
+// https://issues.dlang.org/show_bug.cgi?id=22626
+// REQUIRED_ARGS: -preview=nosharedaccess
+
+shared int k;
+
+class Oops
+{
+    shared int a;
+    shared int* pa;
+    synchronized void oops()
+    {
+        a = 2;
+        static assert (!__traits(compiles, *pa = 2));
+        static assert (!__traits(compiles, k = 2));
```

Comment:
I think a short summary comment of this thread explaining the conclusions should be added, since the linked issue, while it correctly describes undesired behaviour, doesn't comment on the adjacent issues disucssed.